internal/gps/result.go

```diff
@@ -51,11 +52,15 @@ type solution struct {
 // It requires a SourceManager to do the work, and takes a flag indicating
 // whether or not to strip vendor directories contained in the exported
 // dependencies.
-func WriteDepTree(basedir string, l Lock, sm SourceManager, sv bool) error {
+func WriteDepTree(basedir string, l Lock, sm SourceManager, sv bool, logger *log.Logger) error {
 	if l == nil {
 		return fmt.Errorf("must provide non-nil Lock to WriteDepTree")
 	}
 
+	if logger != nil {
```

Comment:
Is it better to define `ctx.VerboseLogger` instead of `Verbose` bool. `Verbose` flag is mostly used as 
```go
if verbose{
  logger.Print ....
}
```